### PR TITLE
doc: nrf: bluetooth: RAS RREQ and RRSP docs

### PR DESCRIPTION
**Pull request by weeTike**
This adds documentation for the Ranging Service Ranging Responder and Ranging Requestor.

### CODE REVIEW

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+Overview
+********
+
+This library implements the Ranging Requestor for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.
+
+Supported features
+==================
+
+- On Demand Ranging Data
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:18]:
```suggestion
The library supports On Demand Ranging Data.
```
The Supported features heading could be left out.

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+Configuration
+*************
+
+Enable RREQ and RRSP kconfig options using the :kconfig:option:`CONFIG_BT_RAS` kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:23]:
```suggestion
To enable the RREQ and RRSP Kconfig options, use :kconfig:option:`CONFIG_BT_RAS`.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+Set the maximum number of antenna paths supported by the device using the :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:25]:
```suggestion
Set the maximum number of antenna paths supported by the device using the :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` Kconfig option.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+This sets the antenna paths per step that can be stored inside the Ranging Service.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:26]:
```suggestion
This sets the antenna paths for each step that can be stored inside the Ranging Service.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+This value must match the supported Channel Sounding capabilities of the device.
+This affects the per-instance memory usage of the Ranging Service.
+
+Set support for storing Mode 3 Channel Sounding steps using the :kconfig:option:`CONFIG_BT_RAS_MODE_3_SUPPORTED` kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:30]:
```suggestion
To set support for storing Mode 3 Channel Sounding steps, use the :kconfig:option:`CONFIG_BT_RAS_MODE_3_SUPPORTED` Kconfig option.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+This will allocate memory for the Ranging Service to store Mode 3 Channel Sounding steps.
+This value must match the supported Channel Sounding capabilities of the device.
+This affects the per-instance memory usage of the Ranging Service.
+
+Enable RREQ kconfig options using the :kconfig:option:`CONFIG_BT_RAS_RREQ` kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:35]:
```suggestion
To enable the RREQ Kconfig options, use :kconfig:option:`CONFIG_BT_RAS_RREQ`.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+Set the number of simultaneously supported RREQ instances using the :kconfig:option:`CONFIG_BT_RAS_RREQ_MAX_ACTIVE_CONN` kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:37]:
```suggestion
To set the number of simultaneously supported RREQ instances, use the :kconfig:option:`CONFIG_BT_RAS_RREQ_MAX_ACTIVE_CONN` Kconfig option.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+
+Set the heap size for the GATT distance measurement module using the :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_RAS_RREQ_GATT_DM` kconfig option.
+
+Set the logging level of the RREQ library using the :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` Kconfig option.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:41]:
```suggestion
To set the heap size for the GATT distance measurement module, use the :kconfig:option:`CONFIG_HEAP_MEM_POOL_ADD_SIZE_RAS_RREQ_GATT_DM` Kconfig option.

To set the logging level of the RREQ library, use the :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` Kconfig option.
```

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+Usage
+*****
+
+The RREQ can be setup as either a Channel Sounding Initiator or Reflector.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:46]:
```suggestion
You can set up the RREQ either as a Channel Sounding Initiator or Reflector.
```

File: doc/nrf/libraries/bluetooth/services/rrsp.rst [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+
+Overview
+********
+
+This library implements the Ranging Responder for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.
+
+Supported features
+==================
+
+- On Demand Ranging Data
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rrsp.rst:18]:
```suggestion
The library supports On Demand Ranging Data.
```
You could leave out the Supported features heading also here.

File: doc/nrf/libraries/bluetooth/services/rrsp.rst [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+Configuration
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rrsp.rst:20]: Do the same adjustments as in the other library doc. This PR also needs a changelog entry.

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/bluetooth/services/rrsp.rst [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+
+Enable RRSP kconfig options using the :kconfig:option:`CONFIG_BT_RAS_RRSP` kconfig option.
+
+Set new connections to be allocated a RRSP instance automatically using the :kconfig:option:`CONFIG_BT_RAS_RRSP_AUTO_ALLOC_INSTANCE` kconfig option.
+
+Set the number of simultaneously supported RRSP instances using the :kconfig:option:`CONFIG_BT_RAS_RRSP_MAX_ACTIVE_CONN` kconfig option.
+
+Set the number of ranging data buffers per connection using the :kconfig:option:`CONFIG_BT_RAS_RRSP_RD_BUFFERS_PER_CONN` kconfig option.
+
+Set the logging level of the RRSP library using the :kconfig:option:`CONFIG_BT_RAS_RRSP_LOG_LEVEL` Kconfig option.
```
Comment (umapraseeda) [inline doc/nrf/libraries/bluetooth/services/rrsp.rst:43]: May be a bulleted list might be beneficial here in this section (in both libraries) :
Example: 
:kconfig:option:`CONFIG_BT_RAS_RRSP_RD_BUFFERS_PER_CONN`  - Sets the number of ranging data buffers per connection.
and so on.

**Review by nordicjm — COMMENTED** [review]
commit should be lowercase for the areas and not have subsys and have the right area e.g. `doc: nrf: bluetooth: Add documentation for RREQ`

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 13]
```diff
@@ -561,6 +561,18 @@ Binary libraries
...
+  * Added
+
+    * Added the Ranging Service Ranging Requestor library for Channel Sounding.
+
+* :ref:`rrsp_readme` library:
+
+  * Added
+
+    * Added the Ranging Service Ranging Responder library for Channel Sounding.
+
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:575]:
```suggestion
* Added the :ref:`rreq_readme` and :ref:`rrsp_readme` libraries.
```
Like this, but remove the changelog entry from this PR and add it as a suggestion to my https://github.com/nrfconnect/sdk-nrf/pull/19240.

File: doc/nrf/libraries/bluetooth/services/rrsp.rst [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+********
+
+This library implements the Ranging Responder for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.
+
+This library supports On Demand Ranging Data.
+
+Configuration
+*************
+
+* :kconfig:option:`CONFIG_BT_RAS` - Enables RREQ and RRSP Kconfig options.
```
Comment (peknis) [inline doc/nrf/libraries/bluetooth/services/rrsp.rst:20]:
```suggestion
To enable this library, use the :kconfig:option:`CONFIG_BT_RAS` Kconfig option.

Check and adjust also the following Kconfig options:
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+
+Configuration
+*************
+
+* :kconfig:option:`CONFIG_BT_RAS` - Enables RREQ and RRSP Kconfig options.
+
+* :kconfig:option:`CONFIG_BT_RAS_MAX_ANTENNA_PATHS` - Sets the maximum number of antenna paths supported by the device.
+  This sets the antenna paths for each step that can be stored inside the Ranging Service.
+  This value must match the supported Channel Sounding capabilities of the device.
+  This affects the per-instance memory usage of the Ranging Service.
```
Comment (sean-madigan) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:25]: This doesn't really apply for rreq. Would maybe just keep the first line

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+
+* :kconfig:option:`CONFIG_BT_RAS_RREQ_LOG_LEVEL` - Sets the logging level of the RREQ library.
+
+Usage
+*****
+
+You can set up the RREQ either as a Channel Sounding Initiator or Reflector.
+
+| See the sample: :file:`samples/bluetooth/channel_sounding_ras_initiator`
+| See the sample: :file:`samples/bluetooth/channel_sounding_ras_reflector`
```
Comment (sean-madigan) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:46]:
```suggestion
```

Only top sample is valid

**Review by peknis — APPROVED** [review]

File: doc/nrf/libraries/bluetooth/services/rreq.rst [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+********
+
+This library implements the Ranging Requestor for Channel Sounding with the corresponding set of characteristics defined in the `Ranging Service Specification`_ and the `Ranging Profile Specification`_.
+
+This library supports On Demand Ranging Data.
+
+Configuration
+*************
+
+To enable this library, use the :kconfig:option:`CONFIG_BT_RAS` Kconfig option.
```
Comment (sean-madigan) [inline doc/nrf/libraries/bluetooth/services/rreq.rst:20]:
```suggestion
To enable this library, use the :kconfig:option:`CONFIG_BT_RAS` and :kconfig:option:`CONFIG_BT_RAS_RREQ` Kconfig options.
```
Maybe this makes more sense

**Review by carlescufi — APPROVED** [review]

**Pull request merged**